azure-pipelines/publishing/github-release.yml

```diff
@@ -9,96 +9,109 @@ pr:
 pool: dotestlab
 
 variables:
-  release.title: $(release_name)
-  release.version: $(release_version)
-# This variable is used later to generate the changelog
-  release.previous_version: $(previous_release_version)
+  Release.Version: $(release_version)
+  Release.PreviousVersion: $(previous_release_version) # Select the previous version, used to generate changelog
+  Release.Title: $(title) # title of the release, following <Release.Version> (YYYY-MM-DD) format i.e. 0.6.0 (2021-03-02)
 
-jobs:
-- job: doclient_release
-  steps:
-  - task: CmdLine@2
-    inputs:
-      script: 'rm -rf /tmp/do_publishing'
-    displayName: 'Clear publishing directory'
+parameters:
+  - name: SkipPublishing
+    displayName: Build only (skip release stage)
+    type: boolean
+    default: false
```

Comment:
>false [](start = 13, length = 5)

I'd keep default as true to avoid publishing accidentally. We want this pipeline to be a fully thought out execution.

Reply:
Well, I'd expect running this pipeline at all to be with the intention of publishing. So this pipeline does two things:
Stage 1: Builds release versions of all our components for all platforms we support
Stage 2: Publishes

Do we see the use case of this pipeline being more for publishing or for validating a build of all our release pipelines? I imagine we won't be using this pipeline too regularly as a 'master' build of all our components, but rather as a 'we are ready to release'. We already have CI via our pipelines running on every check-in on the main branch.

---
In reply to: [586658460](https://github.com/microsoft/do-client/pull/7#discussion_r586658460) [](ancestors = 586658460)

Reply:
Okay, can you remove the default in that case? I'd like it to be explicitly chosen when kicking off the pipeline.

---
In reply to: [586683419](https://github.com/microsoft/do-client/pull/7#discussion_r586683419) [](ancestors = 586683419,586658460)

Reply:
Sure, removed

---
In reply to: [589704917](https://github.com/microsoft/do-client/pull/7#discussion_r589704917) [](ancestors = 589704917,586683419,586658460)